File: Bedu/Java/Clase3/almacenamiento/app/src/main/java/com/charlye934/almacenamiento/MainActivity.java
package com.charlye934.almacenamiento;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    SharedPreferences sharePref;
    Button btnBorrar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        btnBorrar = (Button)findViewById(R.id.btnBorrar);
        btnBorrar.setOnClickListener(
                new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        borrar();
                    }
                }
        );

        sharePref = getApplicationContext().getSharedPreferences(getString(R.string.shared_llave), Context.MODE_PRIVATE);
        if(leerPreferencias()){
            Toast.makeText(getApplicationContext(), "Ya iniciado", Toast.LENGTH_SHORT).show();
        }else{
            SharedPreferences.Editor editor = sharePref.edit();
            editor.putString(getString(R.string.shared_llave_correo),"charlye@gmail.com");
            editor.commit();
            Toast.makeText(getApplicationContext(), "Guardando nuevo registro", Toast.LENGTH_SHORT).show();
        }
    }

    Boolean leerPreferencias(){
        String correo = sharePref.getString(getString(R.string.shared_llave_correo),null);
        return correo != null;
    }

    void borrar(){
        sharePref.edit().clear().commit();
    }
}

File: Bedu/kotlin/Clase1-3/almacenamiento/app/src/main/java/com/charlye934/almacenamientokotlin/MainActivity.kt
package com.charlye934.almacenamientokotlin

import android.content.Context
import android.content.SharedPreferences
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var sharePreferences:SharedPreferences = applicationContext.getSharedPreferences("llave", Context.MODE_PRIVATE)
        btnBorrar.setOnClickListener { sharePreferences.edit().clear().commit() }

        if(leerPreferencias(sharePreferences)){
            Toast.makeText(applicationContext,"ya iniciado",Toast.LENGTH_SHORT).show()
        }else{
            var editor: SharedPreferences.Editor = sharePreferences.edit()
            editor.putString("llaveCorreo","charlye934").commit()
            Toast.makeText(getApplicationContext(), "Guardando nuevo registro", Toast.LENGTH_SHORT).show();
        }
    }

    fun leerPreferencias(sharePreferences:SharedPreferences):Boolean{
        var correo = sharePreferences.getString("llaveCorreo",null)
        return correo != null
    }

}
